cql3/functions/functions.hh

```diff
@@ -33,6 +38,11 @@ namespace functions {
 class functions {
     using declared_t = cql3::functions::declared_t;
     static thread_local declared_t _declared;
+    using filter_pred = std::function<bool(const std::pair<const function_name, shared_ptr<function>>&)>;
+    template<typename F>
+    using it_transformer = std::function<F*(const std::pair<const function_name, shared_ptr<function>>&)>;
+    template<typename F>
+    using filtered_transformed_it = boost::transform_iterator<it_transformer<F>, boost::filter_iterator<filter_pred, declared_t::iterator>>;
```

Comment:
It's better to use type-erased versions, or even return a container. The iterator will break if we yield while generating a result and someone adds a function while we do.

Reply:
Type-erasing sacrifices speed, but speed is not so important here. Returning templated iterators slows compilation down. Anyway returning a container is much safer.